### ISSUE TITLE
🤗 [Question]: How to do Sync on fiberzap? Is that necessary?

### ISSUE THREAD

**Issue by budiprihhastomo** [gofiber/contrib#1160]
### Question Description

I want to use zap on my go app, but as I understand it, zap creates a Sync method, which is used to flushes the buffer into storage. And it is usually done when the application is exited or done periodically. This will ensure the log on buffer will be written to storage. 

Do I need to do it? And where can I do it in fiberzap?

### Checklist:

- [X] I agree to follow Fiber's [Code of Conduct](https://github.com/gofiber/fiber/blob/master/.github/CODE_OF_CONDUCT.md).
- [X] I have checked for existing issues that describe my questions prior to opening this one.
- [X] I understand that improperly formatted questions may be closed without explanation.

**Comment by ReneWerner87** [comment]
@Jamess-Lucass @efectn can you help here

**Comment by Jamess-Lucass** [comment]
@budiprihhastomo Fiberzap simply uses the existing logger that has been created as part of any zap configuration. You can optionally call `.Sync()` throughout your application if required. Does the question come from that fact the example provided in the fiberzap README does not contain `defer logger.Sync()` which is something seen throughout the zap documentation? That hasn't been excluded from the fiberzap documentation on purpose.

https://github.com/gofiber/contrib/tree/main/fiberzap#example
```go
func main() {
    app := fiber.New()
    logger, _ := zap.NewProduction()
    defer logger.Sync() // <- Added this line, which is included in the zap documentation but not fiberzap.

    app.Use(fiberzap.New(fiberzap.Config{
        Logger: logger,
    }))

    app.Get("/", func (c *fiber.Ctx) error {
        return c.SendString("Hello, World!")
    })

    log.Fatal(app.Listen(":3000"))
}
```

The sync functionality is still handled outside of fiberzap, by the core zap library.

I hope this has answered your question, if this hasn't and I've misunderstood your question then please get back to me, a code example might prove helpful if that is the case.

**Comment by budiprihhastomo** [comment]
Thank you, your code example and explanation have completed my question. Maybe this is just an idea that crossed my mind, is it possible to do this synchronization (`logger.Sync()`) )when the response has arrived and been received by the client? 

Maybe I thought of putting it there and making sure that the log is completely flushed. Because if we've to wait when the application got exited, I think there will be a possibility of an excess stack / something that we cannot be sure of.

So my question could also be about, is the event/hook already available on gofiber? I've searched for it but haven't found it.

**Comment by gaby** [comment]
@budiprihhastomo @Jamess-Lucass Yes you need to call `logger.Sync()` if your logger is going to store data. I think for STDOUT it doesn't matter.

That method is also known to not work inside Docker.

I will update the docs.
